tests/unit/backend/modules/webhook/webhook.service.spec.ts

```diff
@@ -0,0 +1,222 @@
+import axios, { AxiosError, AxiosResponse } from 'axios'
+import { ObjectID } from 'bson'
+import mongoose from 'mongoose'
+import dbHandler from 'tests/unit/backend/helpers/jest-db'
+import { mocked } from 'ts-jest/utils'
+
+import getFormModel from 'src/app/models/form.server.model'
+import { getEncryptSubmissionModel } from 'src/app/models/submission.server.model'
+import { WebhookValidationError } from 'src/app/modules/webhook/webhook.errors'
+import { pushData } from 'src/app/modules/webhook/webhook.service'
+import { validateWebhookUrl } from 'src/app/modules/webhook/webhook.utils'
+import formsgSdk from 'src/config/formsg-sdk'
+import { ResponseMode } from 'src/types'
+
+const Form = getFormModel(mongoose)
+const EncryptSubmission = getEncryptSubmissionModel(mongoose)
+
+// Define constants
+const MOCK_ADMIN_OBJ_ID = new ObjectID()
+const MOCK_WEBHOOK_URL = 'https://form.gov.sg/endpoint'
+const ERROR_MSG = 'test-message'
+const MOCK_SUCCESS_RESPONSE = {
+  data: {
+    result: 'test-result',
+  },
+  status: 200,
+  statusText: 'success',
+  headers: {},
+  config: {},
+}
+const MOCK_FAILURE_RESPONSE = {
+  data: {
+    result: 'test-result',
+  },
+  status: 400,
+  statusText: 'failed',
+  headers: {},
+  config: {},
+}
+const MOCK_STRINGIFIED_SUCCESS_RESPONSE = {
+  data: '{"result":"test-result"}',
+  status: 200,
+  statusText: 'success',
+  headers: '{}',
+}
+const MOCK_STRINGIFIED_FAILURE_RESPONSE = {
+  data: `{"result":"test-result"}`,
+  status: 400,
+  statusText: 'failed',
+  headers: '{}',
+}
+const MOCK_EPOCH = 1487076708000
+
+// Set up mocks
+jest.mock('axios')
+const mockAxios = mocked(axios, true)
+jest.mock('src/app/modules/webhook/webhook.utils')
+const mockValidateWebhookUrl = mocked(validateWebhookUrl, true)
+jest.spyOn(Date, 'now').mockImplementation(() => MOCK_EPOCH)
+
+describe('WebhooksService', () => {
+  beforeAll(async () => await dbHandler.connect())
+  afterEach(async () => {
+    await dbHandler.clearDatabase()
+  })
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  let testEncryptSubmission
```

Comment:
add typings to all of these, they will error when we turn on strict mode soon :(

Reply:
yes sir!